tools/shared/src/main/scala/org/scalajs/core/tools/linker/frontend/BaseLinker.scala

```diff
@@ -337,7 +337,8 @@ final class BaseLinker(semantics: Semantics, esLevel: ESLevel, considerPositions
       analysis: Analysis): MethodDef = {
     val encodedName = methodInfo.encodedName
 
-    val targetMDef = findInheritedMethodDef(classInfo, targetName, getTree)
+    val targetMDef = findInheritedMethodDef(analysis, classInfo, targetName,
+        getTree)
```

Comment:
See my other comment. The reflective proxy only knows a method *name*. It does not know in which class/interface it is actually implemented. And that means, at this point, it can resolve to a synthetic default bridge.